src/lab1/Main.java

```diff
@@ -0,0 +1,176 @@
+                    movies = addMovies(numOfMovies, sc, movies);
+                    break;
+                case 4:
+                    System.out.println("The average rating of the movies: " + calcAvgRating(movies));
+                    findMaxAndMinRating(movies);
+                    break;
+                case 5:
+                    System.out.println("Enter the movie name: ");
+                    sc.nextLine();
+                    String name = sc.nextLine();
+                    searchingMovie(name, movies);
+                    break;
+                case 6:
+                    System.out.println("Enter the name of the movie whose rating you want to update: ");
+                    sc.nextLine();
+                    name = sc.nextLine();
+                    updateRatings(name, movies, sc);
+                    break;
+                case 7:
+                    System.out.println("Enter the name of movie that will be deleted: ");
+                    sc.nextLine();
+                    name = sc.nextLine();
+                    movies = deleteMovie(name, movies);
+                    break;
+                case 8:
+                    movies = sortMovies(movies);
+                    System.out.println("List sorted!!");
+                    break;
+                case 0:
+                    return;
+                default:
+                    System.out.println("Invalid input!");
+                    break;
+
+            }
+        }
+    }
+
+    static void searchingMovie(String name, Movie[] movies) {
+        for(Movie movie : movies) {
+            if(movie.getName().equals(name)) {
+                System.out.println("Rating of the searching movie: " + movie.getRating());
+                return;
+            }
+        }
+        System.out.println("Movie is not found!");
+    }
+
+    static void updateRatings(String name, Movie[] movies, Scanner sc) {
+        for(Movie movie: movies) {
+            if(movie.getName().equals(name)) {
+                System.out.println("Enter the updating rating of this movie: ");
+                movie.setRating(sc.nextDouble());
+                return;
+            }
+        }
+        System.out.println("Movie is not found!");
+    }
+
+    static Movie[] deleteMovie(String movieName,Movie[] movies) {
+        boolean found = false;
+        for (Movie movie : movies) {
+            if (movie.getName().equals(movieName)) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            System.out.println("Movie is not found");
+            return movies;
+        } else {
+            System.out.println("Movie is deleted");
+        }
+        Movie[] newMovies = new Movie[movies.length - 1];
+        for(int i = 0, j = 0; i < movies.length; i++, j++) {
+            if(movies[i].getName().equals(movieName)) {
+                j--;
+                continue;
+            }
+            newMovies[j] = movies[i];
+        }
```

Comment:
instead this for loop you can use `System.arraycopy()` it is `@IntrinsicCandidate` (if you don't know what it is pls read about it)